### ISSUE TITLE
Support AWS endpoint overrides via configuration values

### ISSUE THREAD

**Issue by sskrla** [micronaut-projects/micronaut-aws#790]
It is occasionally useful to override the endpoint of AWS clients, especially for testing and Fedramp environments.

Specifically, tests similar to the following would benefit from an endpoint configuration value
```
@MicronautTest
@Testcontainers(disabledWithoutDocker = true)
@TestInstance(PER_CLASS)
internal class S3StorageTest : TestPropertyProvider {
    @Inject
    lateinit var storage: S3Storage

    @Container
    val localstack = LocalStackContainer(LocalStackImage)
        .withServices(S3)
        .also { it.start() }

    @Test fun `stores file`(@TempDir scratch: File) {
        val testFile = File(scratch, "${UUID.randomUUID()}.avro")
        testFile.writeText(randomAscii(1024))
        runBlocking {
            storage.store("test.avro", testFile)
        }
    }

    override fun getProperties() = mapOf(
        "aws.s3.endpoint" to localstack.getEndpointOverride(S3).toString(),
    )
}
```

**Comment by driverpt** [comment]
Any ETA on this one ?

**Comment by alvarosanchez** [comment]
This was done for S3 in #1420

**Comment by scprek** [comment]
Potential fix in https://github.com/micronaut-projects/micronaut-aws/pull/1456 waiting review. @sskrla

**Comment by sskrla** [comment]
> Potential fix in #1456 waiting review. @sskrla

Looks great, and like you out quite a lot of effort into it, thanks for that!

**Comment by scprek** [comment]
@alvarosanchez @sskrla can close this issue?
